core/models/user.py

```diff
@@ -22,15 +21,26 @@ def provider_ids(self):
 
     def user_quota(self):
         identity = self.select_identity()
-        identity_member = identity.identitymembership_set.all()[0]
+        identity_member = identity.identity_memberships.all()[0]
         return identity_member.quota
 
+    @property
+    def current_identities(self):
+        from core.models import Identity
+        all_identities = Identity.objects.none()
+        for group in self.group_set.all():
+            all_identities |= group.current_identities.all()
+        return all_identities
+
+    def can_use_identity(self, identity_id):
+        return self.current_identities.filter(id=identity_id).count() > 0
+
     def select_identity(self):
         """
         Set, save and return an active selected_identity for the user.
         """
         # Return previously selected identity
-        if self.selected_identity and self.selected_identity.is_active():
+        if self.selected_identity and self.selected_identity.is_active(self):
```

Comment:
Hey - we talked about this - I think the keyword argument helps make this _more_ understood (bad English :us: > :gb:)

```python
self.selected_identity.is_active(user=self)
```